include/swift/AST/TypeTransform.h

```diff
@@ -333,11 +364,35 @@ case TypeKind::Id:
         transErrorResult = result;
       }
 
-      if (!changed) return t;
+      if (!changed) {
+        return t;
+      }
+      
+      // Lifetime dependencies get eliminated if their target type was
+      // substituted with an escapable type.
+      auto extInfo = fnTy->getExtInfo();
+      SmallVector<LifetimeDependenceInfo, 2> substDependenceInfos;
+      bool didRemoveLifetimeDependencies
+        = filterEscapableLifetimeDependencies(GenericSignature(),
```

Comment:
Type::transform() is a very hot code path. Can you think of a way to skip all of this new work if the function type does not have lifetime dependencies?

Reply:
`filterEscapableLifetimeDependencies` doesn't do anything if the input list is empty, since it just runs in a `for` loop. I can hoist that check out at each use point if you think it'll make a difference.